astropy/io/fits/hdu/hdulist.py

```diff
@@ -689,6 +698,13 @@ def writeto(self, fileobj, output_verify='exception', clobber=False,
             hdu._prewriteto(checksum=checksum)
             try:
                 hdu._writeto(hdulist._file)
+            except IOError, e:
+                free_space = get_free_space_in_dir(os.getcwd())
+                hdulist_size = np.sum(hdu.size for hdu in self)
+                if not fileobj.file_like and free_space < hdulist_size:
+                    raise IOError("Not enough space on disk. " + str(e))
+                else:
+                    raise IOError(str(e))
```

Comment:
Oh, one other note on
```python
else:
    raise IOError(str(e))
```
since you're just re-raising an `IOError` with the same exception, you can write:

```
else:
    raise
```
rather than creating a new exception.  In fact this is preferable since it makes it easier for Python to generate a full traceback for the exception.  In other case it makes sense to raise a new exception since you're adding a new message.